src/Symfony/Component/Config/Definition/Builder/ChildrenAwareNodeDefinitionInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Definition\Builder;
+
+/**
+ * An interface that must be implemented by nodes which can have children.
+ */
+interface ChildrenAwareNodeDefinitionInterface extends ParentNodeDefinitionInterface
+{
+    /**
+     * @return NodeDefinition[]
+     */
+    public function getChildrenNodeDefinitions();
```

Comment:
This reads weird to me, what about `getChildNodeDefinitions()` or `getChildNodesDefinitions()`?

Reply:
I wonder if we really need this interface or if we shouldn't rather check for the method to exist. We could then add this method to the `ParentNodeDefinitionInterface` in 4.0.

Reply:
I don't have strong opinion about how to name this method. Actually, I already had a hard time finding this one. The same is true for the whole interface.
In case checking for the method to exist, how would you mark deprecation notice though?